src/Appwrite/Usage/Calculators/TimeSeries.php

```diff
@@ -422,36 +423,39 @@ public function __construct(string $region, Database $database, InfluxDatabase $
      */
     private function createOrUpdateMetric(string $projectId, string $time, string $period, string $metric, int $value, int $type): void
     {
+
         $id = \md5("{$time}_{$period}_{$metric}");
         $project = $this->database->getDocument('projects', $projectId);
         $database = call_user_func($this->getProjectDB, $project);
 
-        try {
-            $document = $database->getDocument('stats', $id);
-            if ($document->isEmpty()) {
-                $database->createDocument('stats', new Document([
+        Authorization::skip(function () use ($database, $id, $period, $time, $metric, $value, $type, $projectId) {
+            try {
+                $document = $database->getDocument('stats', $id);
+                if ($document->isEmpty()) {
+                    $database->createDocument('stats', new Document([
                     '$id' => $id,
                     'period' => $period,
                     'time' => $time,
                     'metric' => $metric,
                     'value' => $value,
                     'type' => $type,
                     'region' => $this->region,
```

Comment:
Let's fix indentation here

Reply:
That's a git display thing.
If you look at the file via idea, it is okay.